opencensus/stats/aggregation_data.py

```diff
@@ -52,16 +52,21 @@ def to_point(self, timestamp):
         raise NotImplementedError  # pragma: NO COVER
 
 
-class SumAggregationDataFloat(BaseAggregationData):
+class SumAggregationData(BaseAggregationData):
```

Comment:
Why the decision to make `value_type` an instance attr instead of keeping separate aggregation classes for each value type here?

Reply:
Initially, I had this private class with `value_type` and then bound classes like:
```
SumAggregationDataFloat = functools.partial(SumAggregationData, value_type=value.ValueDouble)
SumAggregationDataInt = functools.partial(SumAggregationData, value_type=value.ValueInt)
```

But then in `SumAggregation.new_aggregation_data`, we needed an additional set of conditionals to check the type of the measure or metric and decide which to use. Instead doing it this way, we just need the one set of conditionals in .`get_metric_type` which we need anyway, and can reuse `MetricDescriptor.to_type_class`. It seemed a little cleaner, but really it isn't that different.

We could also duplicate the classes entirely, instead of the `functools.partial`, but that seemed unnecessary.

I guess the other option is to force the user to decide and provide Int or Float, but since we can determine it that seems easier and less error prone.